starter-code/ShoppingListWithDetailView/app/src/main/java/ly/generalassemb/drewmahrt/shoppinglistdetailview/MainActivity.java

```diff
@@ -20,16 +21,22 @@ protected void onCreate(Bundle savedInstanceState) {
         DBAssetHelper dbSetup = new DBAssetHelper(MainActivity.this);
         dbSetup.getReadableDatabase();
 
-        //Setup the RecyclerView
-        RecyclerView shoppingListRecyclerView = (RecyclerView) findViewById(R.id.shopping_list_recyclerview);
+        Fragment shoppingListFragment = ShoppingListFragment.newInstance(null, this);
 
-        ShoppingSQLiteOpenHelper db = ShoppingSQLiteOpenHelper.getInstance(this);
-        List<ShoppingItem> shoppingList = db.getShoppingList();
-
-        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this,LinearLayoutManager.VERTICAL,false);
-
-        shoppingListRecyclerView.setLayoutManager(linearLayoutManager);
-        shoppingListRecyclerView.setAdapter(new ShoppingListAdapter(shoppingList));
+        getSupportFragmentManager().beginTransaction()
+                .add(R.id.fragment_container, shoppingListFragment).commit();
+    }
 
+    //Change to accept a shopping item as parameter and take values for bundle from shopping item
+    @Override
+    public void onItemSelected(ShoppingItem selectedItem) {
+        Bundle bundle = new Bundle();
+        bundle.putString("NAME", selectedItem.getName());
+        bundle.putString("DESC", selectedItem.getDescription());
+        bundle.putString("PRICE", selectedItem.getPrice());
+        bundle.putString("TYPE", selectedItem.getType());
```

Comment:
One of the requirements was "Gather the details for the selected item from the database (not passed through the intent)". Technically, you're not passing these values via intent, but it's almost the same thing. This works OK in this case, but what if the item had 100 fields of data? Then you'd want to pass just the item ID here, and make a separate database query from the detail fragment to get the one item you need.